crates/core/tedge_mapper/src/c8y/alarm_converter.rs

```diff
@@ -47,10 +47,16 @@ impl AlarmConverter {
             }
             Self::Synced => {
                 //Regular conversion phase
-                let tedge_alarm = ThinEdgeAlarm::try_from(
-                    input_message.topic.name.as_str(),
-                    input_message.payload_str()?,
-                )?;
+                let mqtt_payload = input_message.payload_str()?;
+                let mqtt_topic = input_message.topic.name.clone();
+                let tedge_alarm =
+                    ThinEdgeAlarm::try_from(&mqtt_topic, mqtt_payload).map_err(|e| {
+                        ThinEdgeJsonDeserializerError::FailedToParseJsonPayload {
+                            topic: mqtt_topic,
+                            error: e.to_string(),
+                            payload: mqtt_payload[0..50].into(),
```

Comment:
This panics when the slice end cuts the payload on a non-utf8 boundary.

Reply:
Use: `mqtt_payload.chars.take(50).collect()`.

Reply:
Fixed